django_cte/meta.py

```diff
@@ -47,12 +48,36 @@ def _ref(self):
     def target(self):
         return self._ref.target
 
+    @property
+    def _output_field_or_none(self):
```

Comment:
Where is this used?

Reply:
This is called from  django.db.models.expressions.BaseExpression.get_source_fields
and is a replacement function for django.db.models.expressions.BaseExpression._output_field_or_none
but ensures that we dont try to fetch the field type of a column without a query.